src/main/java/de/tum/in/www1/artemis/web/websocket/team/ParticipationTeamWebsocketService.java

```diff
@@ -197,11 +214,35 @@ else if (submission instanceof TextSubmission textSubmission && exercise instanc
             throw new IllegalArgumentException("Submission type '" + submission.getType() + "' not allowed.");
         }
 
+        if (syncTeammates) {
+            // update the last action date for the user and send out list of team members
+            updateValue(lastActionTracker, participationId, principal.getName());
+            sendOnlineTeamStudents(participationId);
+
+            SubmissionSyncPayload payload = new SubmissionSyncPayload(submission, user);
+            websocketMessagingService.sendMessage(getDestination(participationId, topicPath), payload);
+        }
+    }
+
+    private void patchSubmission(@DestinationVariable Long participationId, @Payload SubmissionPatch submissionPatch, Principal principal, String topicPath) {
+        // Without this, custom jpa repository methods don't work in websocket channel.
+        SecurityUtils.setAuthorizationObject();
+
+        final StudentParticipation participation = studentParticipationRepository.findByIdWithEagerTeamStudentsElseThrow(participationId);
+
+        // user must belong to the team who owns the participation in order to update a submission
+        if (!participation.isOwnedBy(principal.getName())) {
+            return;
+        }
+
+        final User user = userRepository.getUserWithGroupsAndAuthorities(principal.getName());
+        final Exercise exercise = exerciseRepository.findByIdElseThrow(participation.getExercise().getId());
```

Comment:
seems unnecessary. First of all the participation loaded above will include the exercise, second the exercise does not seem to be used. Same for the user, do you really need the full user in the SubmissionPatchPayload? see comment above. If the username is enough, you can simply use `principal.getName()`

Reply:
They are unnecessary, thanks for the remark. I removed these queries.